spec/models/post_spec.rb

```diff
@@ -1,56 +1,29 @@
-# : true
-
-# : true
-
 require 'rails_helper'
 
 RSpec.describe Post, type: :model do
-  let(:post) { Post.create }
-
-  context 'Testing with no inputs' do
-    it 'should not be valid' do
-      expect(post).to_not be_valid
-    end
-
-    it 'likes counter must be 0 or greater' do
-      expect(subject.likes_counter).to be >= 0
-      expect(subject.likes_counter).to be_a(Integer)
-    end
-
-    it 'comments counter must be 0 or greater' do
-      expect(subject.comments_counter).to be >= 0
-      expect(subject.comments_counter).to be_a(Integer)
-    end
-
-    it 'return return_five_most_recent_comments' do
-      expect(subject.return_five_most_recent_comments).to eq(subject.comments.order(updated_at: :desc).limit(5))
-    end
+  let(:user) { User.create(name: 'Victoria Tumwebaze', posts_counter: 0) }
+  let(:post) { Post.create(author_id: user.id, title: 'Sample Post', likes_counter: 0, comments_counter: 0) }
+  let!(:comment1) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+  let!(:comment2) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+  let!(:comment3) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+  let!(:comment4) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+  let!(:comment5) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+
+  describe 'validations' do
+    it { should validate_presence_of(:title) }
+    it { should validate_length_of(:title).is_at_most(250) }
+    it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+    it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
```

Comment:
- [ ] You have all test in post_spec failing due to method validate_presence, kindly fix this
![image](https://github.com/Victoria129/blog-app-rails/assets/66097371/aee553df-33bf-4ec8-bf2d-8588970bfccd)